sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -452,6 +452,16 @@ public UnsafeArrayData copy() {
 
   public static UnsafeArrayData fromPrimitiveArray(
        Object arr, int offset, int length, int elementSize) {
+    UnsafeArrayData result = createFreshArray(length, elementSize);
+    final long headerInBytes = calculateHeaderPortionInBytes(length);
+    final long valueRegionInBytes = (long)elementSize * length;
+    final Object data = result.getBaseObject();
```

Comment:
now the data is `Object` instead of `long[]`. Can we duplicate the code for now and think of how to deduplicate them later?